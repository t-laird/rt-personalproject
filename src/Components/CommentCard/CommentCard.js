import React from 'react';
import './CommentCard.css';

const CommentCard = ({ displayTextIndex, value, note, sender, day }) => {

  const displayText = [
    `${value} points for you!`, 
    `You get ${value} points!`, 
    `Blam! ${value} points!`, 
    `${value} points!  Yeah!`, 
    `${value} big ole points!`
  ];

  return (
    <div className="comment-card-component">
      <div className="card-header">
        <h3>{displayText[displayTextIndex]}</h3><h5 className="day-span">{day}</h5>
      </div>
      <p>{note}</p>
      <div className="card-footer">
        <img src={require('../Header/assets/ninja-logo.svg')} alt="ninja-icon" />
        <h4>{sender}</h4>
      </div>
    </div>
  );
};

export default CommentCard;