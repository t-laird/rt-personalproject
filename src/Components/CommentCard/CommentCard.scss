@import '../../Variables.scss';
@font-face {
	font-family: 'Journal-Regular';
	src: url('./assets/Journal-Regular.otf');
}

.comment-card-component {
	background-color: $c6;
  width: 360px;
  box-sizing: border-box;
  overflow: hidden;
  margin-bottom: 30px;
  &:nth-of-type(odd) {
    margin-right: 30px;
  }

	.card-header {
		background-color: $c4; 	
		padding: 14px 0 15px;
		display: flex;
		align-items: center;
		justify-content: space-around;
	}

	.card-footer {
		padding: 20px;
		display: flex;
		align-items: center;
		justify-content: flex-end;
	}

	h3, h4 {
		font-weight: 900;
	}

	h6{
    font-weight: 900;
		color: $c3;
	}

	h5 {
		color: #fff;
		font-size: 16px;
		font-weight: 900;
	}

	p {
		font-family: 'Journal-Regular';
		font-size: 30px;
    padding: 16px 25px;
    color: $c2;
	}

	img {
		width: 30px;
		margin-right: 5px;
	}
}

@media screen and (max-width: 750px) {
  .comment-card-component {
    &:nth-of-type(odd) {
      margin-right: 0;
    }
  }
}

@media screen and (max-width: 375px) {
  .comment-card-component {
    width: 100%;
  }
}