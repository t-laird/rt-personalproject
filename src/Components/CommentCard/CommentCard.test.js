import React from 'react';
import CommentCard from './CommentCard';
import { shallow } from 'enzyme';

describe('Comment card tests', () => {
  it('should match the snapshot', () => {
    const renderedApp = shallow(<CommentCard />);

    expect(renderedApp).toMatchSnapshot();
  });
});