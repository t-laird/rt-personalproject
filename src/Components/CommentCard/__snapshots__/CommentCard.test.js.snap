// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment card tests should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CommentCard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="card-header"
>
          <h6 />
          <h5
                    className="day-span"
          />
</div>,
        <p />,
        <div
          className="card-footer"
>
          <img
                    alt="ninja-icon"
                    src="ninja-logo.svg"
          />
          <h6 />
</div>,
      ],
      "className": "comment-card-component",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h6 />,
            <h5
              className="day-span"
/>,
          ],
          "className": "card-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "h6",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "day-span",
            },
            "ref": null,
            "rendered": null,
            "type": "h5",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="ninja-icon"
              src="ninja-logo.svg"
/>,
            <h6 />,
          ],
          "className": "card-footer",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "ninja-icon",
              "src": "ninja-logo.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "h6",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="card-header"
>
            <h6 />
            <h5
                        className="day-span"
            />
</div>,
          <p />,
          <div
            className="card-footer"
>
            <img
                        alt="ninja-icon"
                        src="ninja-logo.svg"
            />
            <h6 />
</div>,
        ],
        "className": "comment-card-component",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h6 />,
              <h5
                className="day-span"
/>,
            ],
            "className": "card-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "h6",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "day-span",
              },
              "ref": null,
              "rendered": null,
              "type": "h5",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="ninja-icon"
                src="ninja-logo.svg"
/>,
              <h6 />,
            ],
            "className": "card-footer",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "ninja-icon",
                "src": "ninja-logo.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "h6",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
